controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -174,6 +178,37 @@ func (w *Workload) UpdateKubeletConfigMap(ctx context.Context, version semver.Ve
 		return err
 	}
 
+	// In order to avoid using two cgroup managers on the same machine, cgroupfs and systemd cgroup, starting from
+	// 1.21 image builder is going to configure containerd for using systemd cgroup, and the Kubelet configuration should be changed accordingly.
+	if version.GE(semver.MustParse("1.21.0")) {
```

Comment:
This version variable comes from KCP _current_ set version? If so, is it safe to set this variable even when there are kubelets with older versions, for example during an upgrade?

Reply:
It safe, because we are create the kubelet-config-1.21 config map and nodes will pick up the change only after being updated

Reply:
Let's document that